src/Chapter3.hs

```diff
@@ -560,22 +656,38 @@ introducing extra newtypes.
 🕯 HINT: if you complete this task properly, you don't need to change the
     implementation of the "hitPlayer" function at all!
 -}
+
+-- new types
+newtype Health = Health Int
+newtype Armor = Armor Int
+newtype Attack = Attack Int
+newtype Dexterity = Dexterity Int
+newtype Strength = Strength Int
+newtype Damage = Damage    Int
+newtype Defense = Defense Int
+
 data Player = Player
-    { playerHealth    :: Int
-    , playerArmor     :: Int
-    , playerAttack    :: Int
-    , playerDexterity :: Int
-    , playerStrength  :: Int
+    { playerHealth    :: Health
+    , playerArmor     :: Armor
+    , playerAttack    :: Attack
+    , playerDexterity :: Dexterity
+    , playerStrength  :: Strength
     }
 
-calculatePlayerDamage :: Int -> Int -> Int
-calculatePlayerDamage attack strength = attack + strength
+calculatePlayerDamage :: Attack -> Strength -> Damage
+calculatePlayerDamage (Attack attack) (Strength strength) =
+  Damage (attack + strength)
+
+-- (Attack attack) - its using our types.
+-- but this: Damage (attack + strength) - what is it?
```

Comment:
Here you are using the `Damage` constructor of the `Damage` type to create a value of type `Damage` (yes, constructor and the type has the same name). You just adding manual wrapping here :slightly_smiling_face: Your function must return value of type `Damage`, but values `attack` and `strength` have type `Int`. That's why you need to wrap `Int` to return `Damage`